pkg/webhook/gitmanager/init.go

```diff
@@ -39,14 +50,263 @@ type CustomRepo struct {
 	Mutex          sync.Mutex
 }
 
-func SetupRepo(k *KubeClients, mode StorageModeType, dir string) (*CustomRepo, error) {
+func SetupRepo(k8s *K8sClient, mode StorageModeType, dir string) (*CustomRepo, error) {
+	if mode != StorageModeDisk && mode != StorageModeInMemory {
+		tmp := errors.New("mode must be memory(mem) or disk(disk)")
+		klog.ErrorS(tmp, "incorrect mode")
+		return nil, tmp
+	}
```

Comment:
the function returns an error, so typically you don't want to log the error here but at the caller

```
return nil, fmt.Errorf("mode must be memory(mem) or disk(disk), '%s' is not valid", mode)
```